docs: add faq non clustered operator (#1548)
java-operator-sdk/java-operator-sdk: docs/documentation/faq.md

```

Although you might consider using `EventSources`, to handle reconciliation triggering in a smarter
way. 

### Q: How can I run an operator without cluster scope rights?

> By default, JOSDK require access to CR at cluster scope. You may not be granted such

Reviewer: ```suggestion
By default, JOSDK requires access to CRDs at cluster scope. You may not be granted such
```

> By default, JOSDK requires access to CRDs at cluster scope. You may not be granted such

Reviewer 2: Is this true? By default we should not require access to CRDs just CRs.

Reviewer: I guess, it's not true anymore but the javadoc hasn't been updated when the change was made, so it should be updated as well.

> By default, JOSDK requires access to CRDs at cluster scope. You may not be granted such

Author: ```suggestion
By default, JOSDK requires access to CRs at cluster scope. You may not be granted such
```

rights and you will see some error at startup that looks like:

```plain
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://kubernetes.local.svc/apis/mygroup/v1alpha1/mycr. Message: Forbidden! Configured service account doesn't have access. Service account may have been revoked. mycrs.mygroup is forbidden: User "system:serviceaccount:ns:sa" cannot list resource "mycrs" in API group "mygroup" at the cluster scope.
```

To restrict the operator to a set of namesapce, you may override the namespaces watched by a reconciler
at [Reconciler-level configuration](./configuration.md#reconciler-level-configuration):

```java
Operator operator;
Reconciler reconciler;
...
operator.register(reconciler, configOverrider ->
        configOverrider.settingNamespace("mynamespace"));
```

> 

Reviewer: ```suggestion
Note that configuring the watched namespaces can also be done using the `@ControllerConfiguration` annotation.
```

> Furthermore, you may not be able to list CRDs at startup which is required when `checkingCRDAndValidateLocalModel`

Author: ```suggestion

Furthermore, you may not be able to list CRDs at startup which is required when `checkingCRDAndValidateLocalModel`
```

> is `true` (by default). To disable set it to `false` at [Operator-level configuration](./configuration.md#operator-level-configuration):

Author: @metacosm @csviri From the comment above, I understand the operator do not need "list CRD" right by default ? Does it mean `checkingCRDAndValidateLocalModel` is false by default now ?

Reviewer 2: yep the default should be false:
https://github.com/java-operator-sdk/java-operator-sdk/blob/a6299203025f7e4237cd32f43a4308dd66f7e2ac/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java#L68-L68
